### PR TITLE
fix(vscode): clarify partial-line replace_in_file match failures

### LINKED ISSUE #11439: replace_in_file says SEARCH block does not match when displayed text exists as a partial-line match
### What happened?

`replace_in_file` can report that a `SEARCH` block "does not match anything in the file" even though the displayed text visibly exists in the file as a substring / partial-line match.

This is confusing because the failure message shows text that a user can paste into a normal editor search, such as Notepad++, and find immediately.

### Minimal example

File content:

```cpp
            g_audioFeatures[0].dominantFreq = live.dominantFreq;
        }

        // Build tree spheres with CONTINUOUS time + live audio (no jerk)
```

Model-generated `SEARCH` block:

```text
            g_audioFeatures[0].dominantFreq = live.dominantFreq;
        }

        // Build tree spheres
```

Cline reports that the `SEARCH` block does not match anything in the file.

### Why this is misleading

The displayed text does exist in the file as a substring. The actual reason for failure appears to be that `replace_in_file` treats each SEARCH line as newline-terminated exact span text, so the final line is effectively matched as:

```text
        // Build tree spheres\n
```

but the file contains:

```text
        // Build tree spheres with CONTINUOUS time + live audio (no jerk)\n
```

I understand exact-span replacement can be intentional for safety. The bug is that the error message presents this as "does not match anything" instead of explaining that the visible text is only a partial-line match and the line boundary/newline is part of the match.

### Expected behavior

One of:

- Show escaped line endings in failed SEARCH blocks.
- Say that `replace_in_file` requires exact full-line/span matching and that this SEARCH line only matched a prefix of a longer line.
- Show the nearest matching line from the file.
- Avoid saying "does not match anything" when a substring/prefix match does exist.

Optional feature: support an explicit partial-line/fuzzy matching mode, but the immediate bug is the misleading failure message.

### Version / environment

- Cline VS Code extension: 3.89.0
- OS: Windows
- Matching behavior observed from the installed extension and task history.

### Related

This is related to general `replace_in_file` reliability/diff mismatch reports, but the repro here is specifically about the mismatch between the displayed failed SEARCH text and Cline's internal line-boundary/exact-span matching.

### PR DESCRIPTION
**Pull request by rodboev**
### Description
`replace_in_file` currently collapses a partial-line mismatch into the generic "does not match anything in the file" error. That makes it look like the SEARCH text is missing even when it is visible as part of a longer line.

Changes:
- Add a narrow partial-line diagnostic helper in `apps/vscode/src/core/assistant-message/diff.ts`.
- Reuse the same diagnostic in both matcher implementations so V1 and V2 report the clearer exact-span mismatch, including duplicate earlier matches that fall before the current scan index.
- Guard the helper against very short substring coincidences so single-character SEARCH lines stay on the generic no-match path.
- Keep `WriteToFileToolHandler` telemetry bucketing aligned with the new partial-line error text so these failures still count as `search_not_found`.
- Add regression tests for the issue's minimal example, earlier duplicate partial-line matches, and very short substring false positives.

### Test Procedure
1. Run `cd D:\Repos\cline-pr-replace-in-file-partial-line-match\apps\vscode; npm run format:fix` to normalize the touched matcher files before validation.
2. Run `cd D:\Repos\cline-pr-replace-in-file-partial-line-match\apps\vscode; npx cross-env TS_NODE_PROJECT=./tsconfig.unit-test.json mocha --no-config src/core/assistant-message/diff.test.ts --require ts-node/register --require source-map-support/register --require ./src/test/requires.ts --extension ts --grep "partial-line|very short substring" --exit` to verify the issue repro, the earlier-duplicate diagnostic case, and the short-token false-positive guard in both matcher paths.
3. Run `cd D:\Repos\cline-pr-replace-in-file-partial-line-match\apps\vscode; npx cross-env TS_NODE_PROJECT=./tsconfig.unit-test.json mocha --no-config src/core/assistant-message/diff.test.ts --require ts-node/register --require source-map-support/register --require ./src/test/requires.ts --extension ts --exit` to confirm the existing diff matcher cases still pass after the tighter diagnostic helper and throw-site changes.
4. Run `cd D:\Repos\cline-pr-replace-in-file-partial-line-match\apps\vscode; npx cross-env TS_NODE_PROJECT=./tsconfig.unit-test.json mocha --no-config src/core/task/tools/handlers/__tests__/WriteToFileToolHandler.consecutiveMistakeCount.test.ts --require ts-node/register --require source-map-support/register --require ./src/test/requires.ts --extension ts --exit` to verify the touched handler path still behaves correctly around diff-error handling after the telemetry classification update.

### Type of Change
- [x] Bug fix
- [ ] New feature
- [ ] Breaking change

### Pre-flight Checklist
- [x] I have read the contributing guidelines relevant to this change.
- [x] I have tested the scoped change in the worktree.
- [x] I have kept the change limited to the requested files and behavior.

### Screenshots
N/A - backend matcher and telemetry classification change only.

### Additional Notes
The fix preserves exact-span replacement behavior. It only changes the failure message for meaningful partial-line SEARCH mismatches, keeps the existing diff-failure telemetry classification aligned with that clearer wording, and leaves very short incidental substring matches on the generic no-match path.

## Upstream

Closes #11439.
Reported by @martindye.

### CODE REVIEW

File: apps/vscode/src/core/assistant-message/diff.ts [hunk 1]
```diff
@@ -102,6 +102,65 @@ function lineTrimmedFallbackMatch(originalContent: string, searchContent: string
...
+function getSearchMismatchError(originalContent: string, searchContent: string, startIndex: number): Error {
+	const trimmedSearchContent = searchContent.trimEnd()
+	if (hasPartialLineMismatch(originalContent, searchContent, startIndex)) {
+		return new Error(
+			`The SEARCH block:\n${trimmedSearchContent}\n...only matches part of a longer line in the file. SEARCH/REPLACE blocks require an exact line or span match, including line boundaries.`,
+		)
+	}
+
+	return new Error(`The SEARCH block:\n${trimmedSearchContent}\n...does not match anything in the file.`)
+}
```
[the remark this answers is not in the document]
Comment (rodboev) [reply apps/vscode/src/core/assistant-message/diff.ts:165]: Adjusted the diagnostic in `2393429ae` to retry the partial-line scan from the start of the file when the first pass begins after `lastProcessedIndex`, so earlier duplicate regions still surface the clearer mismatch message.

File: apps/vscode/src/core/assistant-message/diff.ts [hunk 2]
```diff
@@ -102,6 +102,65 @@ function lineTrimmedFallbackMatch(originalContent: string, searchContent: string
...
+			const originalTrimmed = originalLines[i + j].trim()
+			const searchTrimmed = searchLines[j].trim()
+
+			if (originalTrimmed === searchTrimmed) {
+				continue
+			}
+
+			if (searchTrimmed && originalTrimmed.includes(searchTrimmed)) {
+				sawPartialLineMatch = true
+				continue
```
[the remark this answers is not in the document]
Comment (rodboev) [reply apps/vscode/src/core/assistant-message/diff.ts]: Also tightened the helper in `2393429ae` so only meaningful substrings participate in the partial-line diagnostic; very short SEARCH lines now stay on the generic no-match path, with regression coverage for both cases.
